src/views/viewer/ChannelView/ChannelView.tsx

```diff
@@ -207,9 +292,65 @@ export const ChannelView: React.FC = () => {
 
 const usePagination = (currentTab: number) => {
   const [currentPage, setCurrentPage] = useState(0)
+  const [currentSearchPage, setCurrentSearchPage] = useState(0)
   // reset the pagination when changing tabs
   useEffect(() => {
     setCurrentPage(0)
   }, [currentTab])
-  return { currentPage, setCurrentPage }
+  useEffect(() => {
+    setCurrentSearchPage(0)
+  }, [currentSearchPage])
+  return { currentPage, setCurrentPage, currentSearchPage, setCurrentSearchPage }
+}
+
+type useSearchVideosParams = {
+  id: string
+}
+const useSearchVideos = ({ id }: useSearchVideosParams) => {
+  const [isSearchInputOpen, setIsSearchingInputOpen] = useState(false)
+  const [searchQuery, setSearchQuery] = useState('')
+
+  const [isSearching, setIsSearching] = useState(false)
+  const searchInputRef = useRef<HTMLInputElement>(null)
+  const [searchVideo, { loading: loadingSearch, data: searchData, error: errorSearch }] = useSearchLazyQuery()
+
+  const search = () => {
+    searchVideo({
+      variables: {
+        text: searchQuery,
+        whereVideo: {
+          isPublic_eq: true,
+          mediaAvailability_eq: AssetAvailability.Accepted,
+          thumbnailPhotoAvailability_eq: AssetAvailability.Accepted,
+          channelId_eq: id,
+        },
+      },
+    })
```

Comment:
This needs to specify a limit. The default for the search is 5 videos so this is how much we will show currently. Also, we need to somehow get items count to enable proper pagination

Reply:
Limit where? the SearchResults component doesn't do this?

Reply:
Default comes from the query node and we will need to supply it inside the search args. You're right that `SearchResults` doesn't do this and it's a bug (https://github.com/Joystream/atlas/issues/1055)

Reply:
hmm, not sure what to do in this PR then? increase the scope to add the limit thing?

Reply:
Well, channel search won't work properly so we need to add the limit. If you manage to fix the regular search at the same time, even better

Reply:
Made limit 100 which is more than enough for now but we might have to properly refactor the search so it gives totalCount and can be paginated properly